contracts/vPair.sol

```diff
@@ -376,7 +376,7 @@ contract vPair is IvPair, vSwapERC20, ReentrancyGuard {
 
         _update(fetchBalance(token0), fetchBalance(token1));
 
-        require(calculateReserveRatio() < maxReserveRatio, 'TBPT'); // reserve amount goes beyond pool threshold
+        require(calculateReserveRatio() <= maxReserveRatio, 'TBPT'); // reserve amount goes beyond pool threshold
```

Comment:
Can we remove this require and check the trade based on your calculation at the beginning of the function?